sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/PaneInfo.java

```diff
@@ -360,7 +387,8 @@ public PaneInfo decode(final InputStream inStream) throws CoderException, IOExce
       byte keyAndTag = (byte) inStream.read();
       PaneInfo base = Preconditions.checkNotNull(BYTE_TO_PANE_INFO.get((byte) (keyAndTag & 0x0F)));
       long index, onTimeIndex;
-      switch (Encoding.fromTag(keyAndTag)) {
+      boolean elementMetadata = (keyAndTag & 0x80) > 0;
+      switch (Encoding.fromTag((byte) (keyAndTag & 0x7F))) {
```

Comment:
Feel free to ignore my codegolfing...
```suggestion
      boolean elementMetadata = keyAndTag < (keyAndTag &= 0x7f);
      switch (Encoding.fromTag(keyAndTag) {
```